modules/cloud-function/main.tf

```diff
@@ -84,14 +84,14 @@ resource "google_cloudfunctions_function" "function" {
   )
 
   dynamic "event_trigger" {
-    for_each = var.trigger_config == null ? [] : [""]
+    for_each = try(var.trigger_config.v1 == null, true) ? [] : [""]
```

Comment:
We usually do this as
`try(var.trigger_config.v1, null) != null ? [""] : []`. It's the same, but this one is more consistent with the rest of the repo.

Or even better, would it make sense to make `trigger_config` non-nullable?

Reply:
I've made `trigger_config` default to `{v1 = null, v2 = null}` and simplified some these conditions. Thanks.